src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1641,7 +1643,14 @@ TargetShardIntervalForModify(Query *query)
 	}
 
 	prunedShardCount = list_length(prunedShardList);
-	if (prunedShardCount != 1)
+	if (updateOrDelete && prunedShardCount > 1 && prunedShardCount == shardCount)
```

Comment:
I don't think prunedShardCount could ever be > 1 if we're dealing with an INSERT since we explicitly construct prunedShardList with at most 1 item above.